### PR TITLE
SWIFT-747 Implement synchronous transactions API

### PR DESCRIPTION
**Pull request by JamesHeppenstall**

### CODE REVIEW

File: Sources/MongoSwiftSync/ClientSession.swift [hunk 1]
```diff
@@ -96,4 +96,45 @@ public final class ClientSession {
...
+     *
+     * - Parameters:
+     *   - options: The options to use when starting this transaction
+     *
+     * - Throws:
+     *   - `CommandError` if an error occurs that prevents the command from executing.
+     *   - `LogicError` if the session already has an in-progress transaction.
+     *   - `LogicError` if `startTransaction` is called on an ended session.
+     */
+    public func startTransaction(_ options: TransactionOptions?) throws {
```
Comment (kmahar) [inline Sources/MongoSwiftSync/ClientSession.swift]: let's provide a default value of `nil` for the options here.

Comment (JamesHeppenstall) [reply Sources/MongoSwiftSync/ClientSession.swift]: Ok, will do!

File: Sources/MongoSwiftSync/ClientSession.swift [hunk 2]
```diff
@@ -96,4 +96,45 @@ public final class ClientSession {
...
+     *
+     * - Parameters:
+     *   - options: The options to use when starting this transaction
+     *
+     * - Throws:
+     *   - `CommandError` if an error occurs that prevents the command from executing.
+     *   - `LogicError` if the session already has an in-progress transaction.
+     *   - `LogicError` if `startTransaction` is called on an ended session.
+     */
+    public func startTransaction(_ options: TransactionOptions? = nil) throws {
```
Comment (patrickfreed) [inline Sources/MongoSwiftSync/ClientSession.swift]: We typically don't omit the label for options parameters, so for consistency with elsewhere in the driver we probably shouldn't here either.

File: Sources/MongoSwiftSync/ClientSession.swift [hunk 3]
```diff
@@ -96,4 +96,45 @@ public final class ClientSession {
     public func advanceOperationTime(to operationTime: Timestamp) {
         self.asyncSession.advanceOperationTime(to: operationTime)
     }
+
+    /**
```
Comment (patrickfreed) [inline Sources/MongoSwiftSync/ClientSession.swift]: ditto comment in other PR about adding `SeeAlso` link to official docs.

Comment (JamesHeppenstall) [reply Sources/MongoSwiftSync/ClientSession.swift]: Done.

File: Sources/MongoSwiftSync/ClientSession.swift (hunk 2 above)
Comment (JamesHeppenstall) [reply Sources/MongoSwiftSync/ClientSession.swift]: Done.

File: Sources/MongoSwiftSync/ClientSession.swift (hunk 1 above)
Comment (patrickfreed) [reply Sources/MongoSwiftSync/ClientSession.swift]: I think this change along with the underscore removal got reverted by mistake.

Comment (JamesHeppenstall) [reply Sources/MongoSwiftSync/ClientSession.swift]: Thanks for catching that - will revert.

**Pull request merged**